airbyte-integrations/connectors/source-mailchimp/unit_tests/test_source.py

```diff
@@ -0,0 +1,66 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+import pytest
+from airbyte_cdk.logger import AirbyteLogger
+from source_mailchimp.source import SourceMailchimp, MailChimpAuthenticator
+
+logger = AirbyteLogger()
+
+
+def test_check_connection_ok(requests_mock, config, data_center):
+    responses = [
+        {"json": [], "status_code": 200},
+    ]
+    requests_mock.register_uri("GET", f"https://{data_center}.api.mailchimp.com/3.0/ping", responses)
+    ok, error_msg = SourceMailchimp().check_connection(logger, config=config)
+
+    assert ok
+    assert not error_msg
+
+
+def test_check_connection_error(config):
+    ok, error_msg = SourceMailchimp().check_connection(logger, config=config)
+
+    assert not ok
+    assert error_msg
```

Comment:
It could be any error message. It's a general exception for the `check_connection` method. Unit test just checks that method returns `not ok` message and representation of this exception.

Reply:
ok, but anyway let's mock it, unit tests should not access network